tests/integration/test_dependency_replacement.py

```diff
@@ -0,0 +1,74 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from os import path
+import tarfile
+
+import utils
+
+
+def test_dependency_replacement(test_env, tmpdir):
+    """
+    Check that proper versions of dependencies were used.
+
+    Process:
+    * Send new request to Cachito API to fetch retrodep with another version of dependency package
+    * Download a bundle archive
+
+    Checks:
+    * Check that the state of request is complete
+    * Check that in the response there is a key “replaces” with dict values which was replaced
+    * Check that dir deps/gomod/pkg/mod/cache/download/github.com/pkg/errors/@v/… contains
+        only the required version
+    * Check that app/go.mod file has replace directive for the specified package
+    """
+    version_replacements = test_env["dep_replacement"]["dependency_replacements"]
+    deps_values = ["name", "type", "version"]
+    dependency_replacements = [dict(zip(deps_values, item)) for item in version_replacements]
+    client = utils.Client(test_env["api_url"], test_env["api_auth_type"])
+    response_created_req = client.create_new_request(
+        payload={
+            "repo": test_env["package"]["repo"],
+            "ref": test_env["package"]["ref"],
+            "pkg_managers": test_env["package"]["pkg_managers"],
+            "dependency_replacements": dependency_replacements,
+        },
+    )
+    response = client.wait_for_complete_request(response_created_req)
+    assert response.data["state"] == "complete"
+
+    replaced_dependencies = [
+        i["replaces"] for i in response.data["dependencies"] if i["replaces"] is not None
+    ]
+    names_replaced_dependencies = sorted([i["name"] for i in replaced_dependencies])
+    supposed_replaced_dependencies = sorted([i["name"] for i in dependency_replacements])
+    assert names_replaced_dependencies == supposed_replaced_dependencies
+
+    file_name = tmpdir.join(f"download_{str(response.id)}")
+    file_name_tar = tmpdir.join(f"download_{str(response.id)}.tar.gz")
+    resp = client.download_bundle(response.id, file_name_tar)
+    assert resp.status == 200
+    assert tarfile.is_tarfile(file_name_tar)
+
+    with tarfile.open(file_name_tar, "r:gz") as tar:
+        tar.extractall(file_name)
+
+    for dependency in dependency_replacements:
+        dep_name = utils.escape_path_go(dependency["name"])
+        dependency_version_file = (
+            f"{file_name}/deps/gomod/pkg/mod/cache/download/{dep_name}/@v/list"
```

Comment:
Could you construct paths using `os.path.join` to make the tests not OS specific?

I filed https://github.com/release-engineering/cachito/pull/148 to fix a previous test.